mobile-browser-based-version/src/helpers/training_script/training-script.js

```diff
@@ -139,3 +140,121 @@ export async function trainingDistributed(
       console.log('Training finished', info.history);
     });
 }
+
+/**
+ * This function is called when we decided to train using personalized. It creates a local and a global model.
+ * 
+ * @param {TFJS} model the TFJS model used for training
+ * @param {Tensor} trainData the training data
+ * @param {Tensor} labels the labels of the data
+ * @param {Number} batchSize the batchsize used for training
+ * @param {Number} validationSplit the validation split used for trainig
+ * @param {Object} modelCompileData compiling data (optimizer, lost, metric ...)
+ * @param {Object} modelTrainData compiling data (containing number of epoch)
+ * @param {*} onEpochBegin function called at the start of each epoch
+ * @param {*} onEpochEnd function called at the end of each epoch
+ * @param {PeerJS} peerjs peerJS object
+
+ */
+export async function trainingDistributedInteroperability(
+  modelId,
+  trainData,
+  labels,
+  epochs,
+  batchSize,
+  validationSplit,
+  modelCompileData,
+  trainingManager,
+  peerjs,
+  recvBuffer,
+  learningRate = null
+) {
+  // shuffle to avoid having the same thing on all peers
+  //var indices = tf.linspace(0, trainData.shape[0]).cast('int32')
+  //tf.util.shuffle(indices)
+  //const xTrain1d = trainData.gather(indices)
+  //const yTrain1d = labels.gather(indices)
+
+  const savedModelPath = 'indexeddb://working_'.concat(modelId);
+  var model = await tf.loadLayersModel(savedModelPath);
+  peerjs.setDataHandler(handleData, recvBuffer);
+  console.log('Training Distributed with Interoperability');
+
+  // Let's try to see what's displayed
+
+  let modelInputSize = model.layers[0].input.shape[1];
+  let modelOutputSize = model.layers[model.layers.length - 1].outputShape[1];
+  // We create our local model as a wrapper of the input model
+  let localModel;
+
+  try {
+    console.log('Loading Local Model from storage');
+    localModel = await tf.loadLayersModel(savedModelPath.concat('_local'));
+    model = localModel.layers[1];
+  } catch (e) {
+    // For now the only personalization is iFedAvg. More personalization options coming soon.
+    console.log('The local model is not defined, creating a new one ' + e);
+    localModel = tf.sequential();
+
+    localModel.add(
+      new InteroperabilityLayer({
+        units: modelInputSize,
+        inputShape: [modelInputSize],
+      })
+    );
+    localModel.add(model);
+    localModel.add(new InteroperabilityLayer({ units: modelOutputSize }));
+  }
+
+  localModel.summary();
```

Comment:
maybe call it personal model instead of local model? otherwise a bit confusing as all models are stored locally

Reply:
I'll make it consistent over the codebase because I think I have been using both at different places.